system/jhtree/jhtree.cpp

```diff
@@ -606,12 +606,45 @@ class CKeyIdAndPos
     bool operator==(const CKeyIdAndPos &other) { return keyId == other.keyId && pos == other.pos; }
 };
 
-class CNodeMapping : public HTMapping<CJHTreeNode, CKeyIdAndPos>
+class CNodeCacheEntry : public CInterface
 {
+private:
+    std::atomic<const CJHTreeNode *> node = nullptr;
 public:
-    CNodeMapping(CKeyIdAndPos &fp, CJHTreeNode &et) : HTMapping<CJHTreeNode, CKeyIdAndPos>(et, fp) { }
+    ~CNodeCacheEntry()
+    {
+        ::Release(node.load());
+    }
+    inline bool isReady() const
+    {
+        return node != nullptr;
```

Comment:
could use node.load().  Also there may be some benefit in using std::memory_order_[acquire/release] to prevent some memory fences.